testscripts/testrunner/setupenv.go

```diff
@@ -60,5 +60,12 @@ func setupCacheHome(env *testscript.Env) error {
 		return err
 	}
 
+	// Golang caches build outputs for reuse in future builds in GOCACHE
+	goBuildCache := filepath.Join(cacheHome, "go-build")
+	env.Setenv("GOCACHE", goBuildCache)
+	if err = os.MkdirAll(goBuildCache, 0755); err != nil {
+		return err
+	}
```

Comment:
without this I get the following error:
```
failed to initialize build cache at /no-home/Library/Caches/go-build: mkdir /no-home: read-only file system
```

its because `GOCACHE` is likely set to `$HOME/Library/Caches/go-build` and testscripts sets `$HOME` to `/no-home`.